src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -0,0 +1,1872 @@
+            var command_params = new MemoryStream();
+            var command_params_writer = new MonoBinaryWriter(command_params);
+            command_params_writer.WriteLong(pointerValues[pointerId].address);
+            command_params_writer.Write(pointerValues[pointerId].typeId);
+            var ret_debugger_cmd_reader = await SendDebuggerAgentCommand(sessionId, (int) CommandSet.POINTER, (int) CmdPointer.GET_VALUE, command_params, token);
+            var varName = pointerValues[pointerId].varName;
+            if (int.TryParse(varName, out _))
+                varName = $"[{varName}]";
+            return await CreateJObjectForVariableValue(sessionId, ret_debugger_cmd_reader, "*" + varName, false, -1, token);
+        }
+        public async Task<JArray> GetPropertiesValuesOfValueType(SessionId sessionId, int valueTypeId, CancellationToken token)
+        {
+            JArray ret = new JArray();
+            var valueType = valueTypes[valueTypeId];
+            var command_params = new MemoryStream();
+            var command_params_writer = new MonoBinaryWriter(command_params);
+            command_params_writer.Write(valueType.typeId);
+            var ret_debugger_cmd_reader = await SendDebuggerAgentCommand(sessionId, (int) CommandSet.TYPE, (int) CmdType.GET_PARENTS, command_params, token);
+            var parentsCount = ret_debugger_cmd_reader.ReadInt32();
+            List<int> typesToGetProperties = new List<int>();
+            typesToGetProperties.Add(valueType.typeId);
+            for (int i = 0 ; i < parentsCount; i++)
+            {
+                typesToGetProperties.Add(ret_debugger_cmd_reader.ReadInt32());
+            }
+            for (int i = 0 ; i < typesToGetProperties.Count; i++)
+            {
+                var properties = await CreateJArrayForProperties(sessionId, typesToGetProperties[i], valueType.valueTypeBuffer, valueType.valueTypeJson, valueType.valueTypeAutoExpand, $"dotnet:valuetype:{valueType.Id}", i == 0, token);
+                ret = new JArray(ret.Union(properties));
+            }
+
+            return ret;
+        }
+
+        public bool AutoExpandable(string className) {
+            if (className == "System.DateTime" ||
+                className == "System.DateTimeOffset" ||
+                className == "System.TimeSpan")
+                return true;
+            return false;
+        }
+
+        public bool AutoInvokeToString(string className) {
+            if (className == "System.DateTime" ||
+                className == "System.DateTimeOffset" ||
+                className == "System.TimeSpan" ||
+                className == "System.Decimal"  ||
+                className == "System.Guid")
+                return true;
+            return false;
+        }
+
+        public async Task<JObject> CreateJObjectForVariableValue(SessionId sessionId, MonoBinaryReader ret_debugger_cmd_reader, string name, bool isOwn, int typeIdFromAttribute, CancellationToken token)
```

Comment:
This method is pretty large and doesn't have a lot of shared state in the switch.